### ISSUE TITLE
Type indexing with string literals starting with __ does not work

### ISSUE THREAD

**Issue by guncha** [Microsoft/TypeScript#14268]
**TypeScript Version:**  2.3.0-dev.20170223

**Code**

It's very likely that this is already reported, but I couldn't find it after a quick search through the issues.

```ts
type Foo = {
  bar: string
  __typename: string
}

type A = Foo["bar"] // ok 
type B = Foo["__typename"] // Property '__typename' does not exist on type 'Foo'.
```

**Expected behavior:**

Type `B` should be `string`.

**Actual behavior:**

Error is thrown.

**Comment by guncha** [comment]
I did find an older similar issue: https://github.com/Microsoft/TypeScript/issues/8614

Lo and behold: 

```ts
type B = Foo["___typename"] // three underscores, ok
```

**Comment by weswigham** [comment]
This should be fixed by #16915
